src/main/java/com/logpresso/scanner/Log4j2Scanner.java

```diff
@@ -941,6 +955,38 @@ protected Status checkLog4jVersion(File jarFile, boolean fix, ZipFile zipFile) t
 		}
 	}
 
+	private Status checkLog4j_12_Version(File jarFile, ZipFile zipFile) throws IOException {
+		ZipEntry entry = zipFile.getEntry(LOG4j_12_CORE_POM_PROPS);
+		if (entry == null) {
+			entry = zipFile.getEntry(LOG4j_12_JMSAPPENDER);
+			if(entry != null) {
+				String path = jarFile.getAbsolutePath();
+				printDetection_12(jarFile, null, POTENTIALLY_VULNERABLE);
+				return Status.POTENTIALLY_VULNERABLE;
+			}
+			return Status.NOT_VULNERABLE;
+		}
+
+		InputStream is = null;
+		try {
+			is = zipFile.getInputStream(entry);
+
+			String version = loadVulnerableLog4jVersion_12(is);
+			if (version != null) {
+				boolean jmsAppender = zipFile.getEntry(LOG4j_12_JMSAPPENDER) != null;
+				if(jmsAppender) {
+					String path = jarFile.getAbsolutePath();
+					printDetection_12(jarFile, null, version);
+				}
+				return jmsAppender ? Status.NOT_VULNERABLE : Status.POTENTIALLY_VULNERABLE;
```

Comment:
It should be `return jmsAppender ? Status.POTENTIALLY_VULNERABLE : Status.NOT_VULNERABLE;`. I will patch it after merge.

Reply:
Sure... Thanks